independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanResolverImpl.java

```diff
@@ -107,6 +104,13 @@ public BeanInfo resolveAmbiguity(Set<BeanInfo> beans) {
         return resolved.isEmpty() ? Collections.emptyList() : resolved;
     }
 
+    Collection<BeanInfo> potentialBeans(Type type) {
+        if (type.kind() == CLASS || type.kind() == PARAMETERIZED_TYPE) {
+            return beanDeployment.getBeansByRawType(type.name());
```

Comment:
So this will return an empty list for `java.lang.Object`, right? Shouldn't we return `beanDeployment.getBeans()` for `java.lang.Object`?

Reply:
Hm,yes you're right that we should return all beans for `java.lang.Object`. In fact, we should always return all beans if no value exists for a specific key.

Reply:
Ah right, correct!

Reply:
I added a special handling for `java.lang.Object` in the end because that's the only type which is handled separately. Hope it's correct now...

Reply:
Agree.